ledger/smt.go

```diff
@@ -0,0 +1,402 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ledger
+
+// The SMT is derived from https://github.com/aergoio/SMT with modifications
```

Comment:
Should we carry that code's copyright banner?

Reply:
I have no idea.  Would we use their banner in lieu of our own, or both banners?  I can do whatever, I'm just not 100% clear on the right thing to do.

Reply:
It is not clear which part has to be fully reviewed / new code and which part is from upstream.